### DISCUSSION TITLE
How to enable GitHub push on self hosted docker?

### DISCUSSION THREAD

**Discussion by WilliamHingston** [WeblateOrg/weblate#5476] [Q&A]
I can't see any way to enable GitHub push on my self hosted docker setup. The push button is disabled and I can't find where to change it in the settings.

I set up GitHub SSH and created my project by pulling from GitHub with:
`docker exec --user weblate <DOCKER_CONTAINER_ID> weblate import_project --file-format po <PROJECT_NAME> <SSH_GIT_REPO> <GIT_BRANCH> <MASK>`

**Comment by nijel** [comment]
The button is disabled when push URL is not configured, see https://docs.weblate.org/en/latest/admin/projects.html#repository-push-url

**Reply by WilliamHingston** [reply]
I've read that page and also https://docs.weblate.org/en/latest/vcs.html#vcs-repos but I still can't find what settings I need to update to enable push for self hosted docker. I would have thought setting up SSH would be enough. Is there some environment variables I need to set in the docker-compose.override.yml?

**Reply by nijel** [reply]
The component needs to have configured push URL  - that's the documentation link I've posted before. The import project command doesn't do that automatically.
